src/server/scripts/Northrend/FrozenHalls/ForgeOfSouls/boss_bronjahm.cpp

```diff
@@ -176,6 +201,8 @@ class boss_bronjahm : public CreatureScript
 
                 DoMeleeAttackIfReady();
             }
+        private:
+            uint32 _soulFragmentsSpawned;
```

Comment:
add _soulFragmentsSpawned = 0; to the constructor or use Initialize() method

check for example this: https://github.com/TrinityCore/TrinityCore/blob/6.x/src/server/scripts/Northrend/DraktharonKeep/boss_trollgore.cpp#L81